pkg/process/checks/net.go

```diff
@@ -148,7 +184,13 @@ func batchConnections(
 			ContainerForPid:   ctrIDForPID,
 			EncodedDNS:        dnsEncoder.Encode(batchDNS),
 			ContainerHostType: cfg.ContainerHostType,
-		})
+		}
+		// only add the telemetry to the first message to prevent double counting
+		if len(batches) == 0 {
+			cc.Telemetry = telemetry
+		}
```

Comment:
Can we add a test to confirm this behavior in `net_test.go`

Reply:
Sure!